assets/app/scripts/controllers/overview.js

```diff
@@ -40,22 +53,156 @@ angular.module('openshiftConsole')
+      console.log("monopodsByService", $scope.monopodsByService); 
+    };
+
+    // Filter out monopods we know we don't want to see
+    var showMonopod = function(pod) {
+      // Hide pods in the Succeeded or Terminated phase since these are run once pods
+      // that are done
+      if (pod.status.phase == 'Succeeded' || pod.status.phase == 'Terminated') {
+        return false;
+      } 
+      // Hide our build pods since we are already showing details for currently running or recently
+      // run builds under the appropriate areas
+      for (var id in $scope.builds) {
+        if ($scope.builds[id].podName == pod.metadata.name) {
+          return false;;
+        }
+      }
+      // Hide our deployer pods since it is obvious the deployment is happening when the new deployment
+      // appears.
+      if (pod.metadata.annotations && pod.metadata.annotations.deployment) {
+        return false;
+      }
+
+      return true;
+    };
+
+    var deploymentConfigsByService = function() {
+      $scope.deploymentConfigsByService = {"": {}};
+      angular.forEach($scope.deploymentConfigs, function(deploymentConfig, depName){
+        var foundMatch = false;
+        var depConfigSelector = new LabelSelector(deploymentConfig.template.controllerTemplate.replicaSelector);
+        angular.forEach($scope.services, function(service, name){
+          $scope.deploymentConfigsByService[name] = $scope.deploymentConfigsByService[name] || {};
+
+          var serviceSelector = new LabelSelector(service.spec.selector);
+          if (serviceSelector.covers(depConfigSelector)) {
+            $scope.deploymentConfigsByService[name][depName] = deploymentConfig;
+            foundMatch = true;
+          }
+        });
+        if (!foundMatch) {
+          $scope.deploymentConfigsByService[""][depName] = deploymentConfig;
+        }
       });
+    };
+
+    var deploymentsByService = function() {
+      $scope.deploymentsByService = {"": {}};
+      $scope.deploymentsByServiceByDeploymentConfig = {"": {}};
+
+      angular.forEach($scope.deployments, function(deployment, depName){
+        var foundMatch = false;
+        var deploymentSelector = new LabelSelector(deployment.spec.selector);
 
-      console.log("podsByServiceByLabel", $scope.podsByServiceByLabel);      
+        angular.forEach($scope.services, function(service, name){
+          $scope.deploymentsByService[name] = $scope.deploymentsByService[name] || {};
+          $scope.deploymentsByServiceByDeploymentConfig[name] = $scope.deploymentsByServiceByDeploymentConfig[name] || {};
+
+          var serviceSelector = new LabelSelector(service.spec.selector);
+          if (serviceSelector.covers(deploymentSelector)) {
+            $scope.deploymentsByService[name][depName] = deployment;
+
+            var depConfigName = "";
+            if (deployment.metadata && deployment.metadata.annotations && deployment.metadata.annotations.deploymentConfig) {
```

Comment:
drop ` && deployment.metadata.annotations.deploymentConfig`, do `|| ""` inside the block